### ISSUE TITLE
Strange thing about RegisterClass: the lpfnWndProc property of WNDCLASS struct

### ISSUE THREAD

**Issue by Meigyoku-Thmn** [thpatch/win32_utf8#1]
At here https://github.com/thpatch/win32_utf8/blob/master/src/user32_dll.c#L45 
I find it's very strange that you just copy the lpfnWndProc without doing anything more. 
A WindowProc callback can receive certain messages (such as [WM_CREATE](https://msdn.microsoft.com/en-us/library/windows/desktop/ms632619(v=vs.85).aspx)) which have string in lParam or wParam. 
So if we pass a WindowProc to RegisterClassW, then it can receive wide character string.
But the WindowProc itself is programmed to handle only ANSI string (or Multibyte string). So how can it work properly?

**Comment by Amaroq-Clearwater** [comment]
That's a good question. The repository was last updated in 2019 though, so this might have already been addressed.

**Comment by brliron** [comment]
Nope, I think that's still the case. The only thing we do is replacing DefWindowProcA with DefWindowProcW.
This project is pushed by the needs of thcrap, which itself is pushed by the Touhou Project games, so I guess these games just don't do that.

**Comment by Amaroq-Clearwater** [comment]
This project should probably be forked then, that way it can be updated more frequently for a more general use case.

**Comment by brliron** [comment]
I don't think a fork would fix this issue. What this project would need, in order to have greater compatibility than whatever the Touhou Project games need, is more contributors.

**Comment by Amaroq-Clearwater** [comment]
Might wanna advertise it then.
And win32_smartdlg.
